plugins/jenkins-common/src/permissions.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Permission } from '@backstage/plugin-permission-common';
+
+/**
+ * @public
+ */
+export const RESOURCE_TYPE_JENKINS = 'jenkins';
+
+/**
+ * This permission is used to determine if a user is allowed to execute an action in jenkins plugin
+ *
+ * @public
+ */
+export const jenkinsExecutePermission: Permission = {
+  name: 'jenkins.execute',
+  attributes: {
+    action: 'update',
+  },
+  resourceType: RESOURCE_TYPE_JENKINS,
```

Comment:
Since this permission is authorized against catalog entities, we can use RESOURCE_TYPE_CATALOG_ENTITY here. Once we do that, the permission backend will automatically start using the catalog-backend to apply any conditions that need to be applied. This makes life a lot simpler - it means that we can remove all the changes to permission-backend in this PR, and everything should still work!

Reply:
I agree the only reason why I created the custom resource type and permissions is that in RFC has been described something like:

>  If your plugin needs authorization, you can define a set of permissions that describe the available actions. It’s recommended that you define all of the permissions for your plugin in a single module (see [Isomorphic packages for permissions](https://github.com/backstage/backstage/pull/9840#isomorphic-packages-for-permissions)).

I thought it could be better to have an independent permissions for a plugin so the client can able to choose or decide to activate or deactivate a permission inside for a specific plugin. (e.g. some of users would like to have re-build even if they enable catalog permissions to prevent deleting an entity) 

But I'm still trying to understand the permission plugin philosophy in backstage, if you think it should be fine to use catalog permissions here, I can remove/clean-up the code to use catalog resources.

Reply:
The permission definitely makes sense! It's the resource type, conditions, router etc. that seem to me to not be needed, as long as we want to authorize this permission in the context of catalog entities. 

If we wanted to instead consider jenkins jobs to be separate resources, and authorize permissions in the context of them instead of catalog entities, then we would need the stuff in permission-backend. The catalog entity approach seems like the right thing to me though :).

Reply:
Then I will use catalogEntityCreatePermission this permission request to check if user has a write access, is it fine for you? :)